### PR TITLE
Player stagger

### LINKED ISSUE #21: Player: taking damage and stagger

### PR DESCRIPTION
**Pull request by henriiquecampos**
Add a Stagger state closes #21 it didn't need an animation, I think the animation itself must be on Skin, the Stagger basically yields for its duration to finish than goes back to the Move/Idle state.

### CODE REVIEW

File: hook!-full-game/src/Areas/SpikeLevelTest.tscn [hunk 1]
```diff
@@ -0,0 +1,125 @@
+[gd_scene load_steps=10 format=2]
```
Comment (NathanLovato) [inline hook!-full-game/src/Areas/SpikeLevelTest.tscn]: You added a spike to LevelTest so we don't need this extra scene

File: hook!-full-game/src/Combat/HitBox/HitBox.gd [hunk 2]
```diff
@@ -10,7 +10,7 @@ func _ready() -> void:
 
 
 func _on_area_entered(damage_source: Area2D) -> void:
-	var hit = HitFactory.calculate_hit(damage_source)
+	var hit = HitFactory.generate_hit(damage_source)
```
Comment (NathanLovato) [inline hook!-full-game/src/Combat/HitBox/HitBox.gd]: good catch!

File: hook!-full-game/src/Player/Player.gd [hunk 3]
```diff
@@ -24,6 +24,7 @@ func _ready() -> void:
 
 func take_damage(source: Hit) -> void:
 	stats.take_damage(source)
+	state_machine.transition_to("Stagger")
```
Comment (NathanLovato) [inline hook!-full-game/src/Player/Player.gd]: Actually, you don't want to do that. You should react to a signal on stats, something like "damage_taken". So we don't introduce bugs if e.g. we add an invincibility buff, an armor spell, something that prevents from taking damage in the future.

I think that taking damage in the hook state shouldn't stagger the character, so I'd put the signal reaction inside the states as well. Mainly on Move for now. You can use the `Events` autoload to connect Move to Stats without having to pass one to the other for now.

File: hook!-full-game/src/Areas/SpikeLevelTest.tscn (hunk 1 above)
Comment (henriiquecampos) [reply hook!-full-game/src/Areas/SpikeLevelTest.tscn]: Ohh sorry, it wasn't meant to commit the current `LevelTest.tscn` :sweat_smile: I duplicated it with the Spikes so it wouldn't cause conflicts. It was meant to push only the `SpikeLevelTest.tscn`

File: hook!-full-game/src/Player/Player.gd (hunk 3 above)
Comment (henriiquecampos) [reply hook!-full-game/src/Player/Player.gd]: I used `_player.stats` instead

**Comment by henriiquecampos** [comment]
Alright, PR updated and clean :wink:

**Comment by NathanLovato** [comment]
Thanks, just fixed conflicts and merged!

**Pull request merged**